### ISSUE TITLE
Improve Windows support

### ISSUE THREAD

**Issue by arithmetric** [phase2/grunt-drupal-tasks#129]
Several issues have been reported with using Grunt Drupal Tasks on Windows (or in some cases inside a folder shared between a Linux VM and a Windows host).

These issues require investigation:
### Support for symlinks

Reports:

I've found ln -s works on windows in git shell but not for directories. You can symlink files but not dirs

Closest you can get is a junction. Mklink.exe /J 
Needs to be run as admin and is a DOS command. (Ie I couldn't get it to run with a term emulator in my bash script)

So basically I would run my build script on windows and manually junction the directories afterwards. Fine until you try to rebuild. Then you find out the difference between a symlink and a hard link

See: http://en.wikipedia.org/wiki/NTFS_junction_point
### Drush make support

Reports: Drush make had a hard time applying patches from /tmp

**Comment by arithmetric** [comment]
Consider falling back to "copy" instead of symlink for Windows machines if the junction approach doesn't work.

**Comment by markdboyd** [comment]
I worked on this a bit the other day. My approach was to detect the OS in Grunt and switch whether to use symlinks or direct copying of the files accordingly.  Here's what I have so far:

https://gist.github.com/markdboyd/2aba928f6fb2e7ab50a7

Note that there may be some tasks in there which are specific to my project, but in general the approach would be the same. I did test it on OS X and the file copying worked if I set as the default, but I haven't had time to set up a Windows environment for testing.
